src/e2e/java/uk/gov/hmcts/reform/sscs/functional/handlers/AdminAppealWithdrawnNotificationsTest.java

```diff
@@ -0,0 +1,104 @@
+
+
+public class AdminAppealWithdrawnNotificationsTest extends AbstractFunctionalTest {
+
+    public AdminAppealWithdrawnNotificationsTest() {
+        super(30);
+    }
+
+    @Rule
+    public Retry retry = new Retry(0);
+
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(60);
+
+    @Before
+    public void setUp() {
+        initialiseCcdCase();
+    }
+
+    @Test
+    @Parameters({
+        "Appellant, 8620e023-f663-477e-a771-9cfad50ee30f, 446c7b23-7342-42e1-adff-b4c367e951cb, 1, 1, 1",
+        "Appointee, 8620e023-f663-477e-a771-9cfad50ee30f, 446c7b23-7342-42e1-adff-b4c367e951cb, 1, 1, 1",
+        "Reps, e29a2275-553f-4e70-97f4-2994c095f281, f59440ee-19ca-4d47-a702-13e9cecaccbd, 1, 1, 2"
+    })
+    public void givenCallbackWithSubscriptions_shouldSendEmailSmsAndLetterNotifications(
+        String subscription,
+        String emailId,
+        String smsId,
+        int expectedNumEmailNotifications,
+        int expectedNumSmsNotifications,
+        int expectedNumLetters) throws Exception {
+
+        simulateCcdCallback(ADMIN_APPEAL_WITHDRAWN, "handlers/" + ADMIN_APPEAL_WITHDRAWN.getId() + subscription
+            + "Callback.json");
+
+        List<Notification> notifications = tryFetchNotificationsForTestCase(emailId, smsId);
+
+        assertEquals(expectedNumEmailNotifications, getNumberOfNotificationsForGivenEmailOrSmsTemplateId(notifications, emailId));
+        assertEquals(expectedNumSmsNotifications, getNumberOfNotificationsForGivenEmailOrSmsTemplateId(notifications, smsId));
+        assertTrue(fetchLetters(expectedNumLetters));
+    }
+
+    private boolean fetchLetters(int expectedNumLetters) {
+        do {
+            if (getNumberOfLetterCorrespondence() == expectedNumLetters) {
+                return true;
+            }
+            delayInSeconds();
+        } while (true);
+    }
+
+    private void initialiseCcdCase() {
+        getCaseData().setCorrespondence(null);
+        getCaseData().setSubscriptions(null);
+        getCcdService().updateCase(getCaseData(), getCaseId(), CASE_UPDATED.getId(), "create by Test",
+            "Notification Service updated case", getIdamTokens());
+    }
+
+    private long getNumberOfNotificationsForGivenEmailOrSmsTemplateId(List<Notification> notifications, String emailId) {
+        return notifications.stream()
+            .filter(notification -> notification.getTemplateId().equals(UUID.fromString(emailId)))
+            .count();
+    }
+
+    private void delayInSeconds() {
+        try {
+            TimeUnit.SECONDS.sleep(5);
+        } catch (InterruptedException ie) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    private long getNumberOfLetterCorrespondence() {
+        List<Correspondence> correspondence = getCcdService()
+            .getByCaseId(getCaseId(), getIdamTokens()).getData().getCorrespondence();
+        if (correspondence == null) {
+            return 0;
+        }
+        return correspondence.stream()
```

Comment:
Is there anything here which checks whether it is the correct letter? One of the problems I addressed last night was that we were getting false positives as the test wasn't checking that the expected notification was actually being received, just that one had been received. I suspect you may have the same problem here but let me know if I'm wrong.

Reply:
ok, yes, I get your point and that is why I initialise the case in CCD before running the test like this: getCaseData().setCorrespondence(null);
so after the test if there is any correspondence it is because it was added as part of that test
since the test created a unique case only for that test
I haven't encountered any issues following this approach